st2api/st2api/controllers/resource.py

```diff
@@ -171,92 +187,87 @@ def _get_all(self, exclude_fields=None, sort=None, offset=0, limit=None, query_o
             else:
                 filters['__'.join(v.split('.'))] = filter_value
 
-        extra = {
-            'filters': filters,
-            'sort': sort,
-            'offset': offset,
-            'limit': limit
-        }
-        LOG.info('GET all %s with filters=%s' % (pecan.request.path, filters), extra=extra)
-
         instances = self.access.query(exclude_fields=exclude_fields, **filters)
         if limit == 1:
             # Perform the filtering on the DB side
             instances = instances.limit(limit)
 
-        if limit:
-            pecan.response.headers['X-Limit'] = str(limit)
-        pecan.response.headers['X-Total-Count'] = str(instances.count())
-
         from_model_kwargs = from_model_kwargs or {}
-        from_model_kwargs.update(self._get_from_model_kwargs_for_request(request=pecan.request))
+        from_model_kwargs.update(self._get_from_model_kwargs_for_request(**kwargs))
 
         result = []
         for instance in instances[offset:eop]:
             item = self.model.from_model(instance, **from_model_kwargs)
             result.append(item)
 
-        return result
+        resp = Response(json=result)
+        resp.headers['X-Total-Count'] = str(instances.count())
+        if limit:
+            resp.headers['X-Limit'] = str(limit)
+
+        return resp
 
-    def _get_one(self, id, exclude_fields=None):
+    def _get_one(self, id, exclude_fields=None, **kwargs):
```

Comment:
I still side with kami. I’ll make a tradeoff on dev productivity there. It won’t be worse than waiting for builds to pass. Seriously, we had numerous bugs in CLI code because of kwargs abuse.